os/pm/pm_timer/pm_timer_activity.c

```diff
@@ -0,0 +1,244 @@
+			leave_critical_section(state);
+		}
+	}
+}
+
+/************************************************************************
+ * Public Functions
+ ************************************************************************/
+
+/****************************************************************************
+ * Name: pm_set_wakeup_timer
+ *
+ * Description:
+ *   This function is called just before sleep to start the required PM wake up
+ *   timer. It will start the first timer from the g_pmTimer_activeList with the
+ *   required delay.(delay should be positive)
+ * 
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   0 - system can go to sleep
+ *   -1 - system should not go to sleep
+ *
+ ****************************************************************************/
+
+int pm_set_wakeup_timer(void)
+{
+	/* if no timer present , do we need to wake up board ??? */
+	if (g_pmTimer_activeList.head == NULL) {
+		pmdbg("the g_pmTimer_activeList list is empty!!!\n");
+		return PM_TIMER_SUCCESS;
+	}
+
+	pm_wakeup_timer_t *curr;
+    	pm_wakeup_timer_t *prev;
+    	prev = curr = (pm_wakeup_timer_t *)g_pmTimer_activeList.head;
+
+	/* Remove the wakeup timers that are already expired. 
+	 * Although this must have already handled by pm_timer_update() */
+	while (curr->delay <= 0 && curr->next) {
+		prev = curr;
+		curr = curr->next;
+	}
+
+	if (curr->delay > 0) {
+		pmdbg("Setting timer and Board will wake up after %d millisecond\n", curr->delay);
+		pm_set_timer(PM_WAKEUP_TIMER, curr->delay * 1000);
+	} 
+	
+	return PM_TIMER_SUCCESS;
+}
+
+/****************************************************************************
+ * Name: pm_timer_update
+ *
+ * Description:
+ *   This function decreases the delay of head pm timer in the 
+ *   g_pmTimer_activeList by 1 after every sys tick. If the delay becomes 0,
+ *   It expires the pm timer.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None.
+ * 
+ * Assumption: This should be also implemented for CONFIG_SCHED_TICKLESS and
+ * CONFIG_SCHED_TICKSUPRESS.
+ *
+ ****************************************************************************/
+
+void pm_timer_update(void)
+{
+	/* Check if there are any active pm timers to process */
+	if (g_pmTimer_activeList.head) {
+
+		/* Decrement the delay of the head pm timer */
+		--(((pm_wakeup_timer_t *)g_pmTimer_activeList.head)->delay);
+
+		/* process the expired pm timers */
+		pm_timer_expire();
+
+	}
+}
+
+/************************************************************************
+ * Name: pm_wakeup_handler
+ *
+ * Description:
+ *   This function is called just after board wakes up from sleep.
+ *   It updates the pm timer delay status in the g_pmTimer_activeList by
+ *   decreasing timer's delay to missing ticks.
+ *
+ * Parameters:
+ *   missing_tick : missing ticks after sleep
+ *
+ * Return Value:
+ *   None
+ *
+ ************************************************************************/
+
+void pm_wakeup_handler(clock_t missing_tick)
```

Comment:
This operation appear to only for update pm timer tick.

How about change function name to pm_timer_update? and merge above fucntion?